Src/FluentAssertions/Equivalency/EnumEqualityStep.cs

```diff
@@ -56,5 +52,59 @@ public bool CanHandle(IEquivalencyValidationContext context, IEquivalencyAsserti
 
             return true;
         }
+
+        private static void HandleByValue(IEquivalencyValidationContext context)
+        {
+            decimal? subjectsUnderlyingValue = ExtractDecimal(context.Subject);
+            decimal? expectationsUnderlyingValue = ExtractDecimal(context.Expectation);
+
+            string subjectsName = EnumDescription(context.Subject, subjectsUnderlyingValue);
+            string expectationName = EnumDescription(context.Expectation, expectationsUnderlyingValue);
+
+            Execute.Assertion
+                .ForCondition(subjectsUnderlyingValue == expectationsUnderlyingValue)
+                .FailWith($"Expected enum to be {expectationName}{{reason}}, but found {subjectsName}.");
+        }
+
+        private static void HandleByName(IEquivalencyValidationContext context)
+        {
+            string subject = context.Subject.ToString();
+            string expected = context.Expectation.ToString();
+
+            decimal? subjectsUnderlyingValue = ExtractDecimal(context.Subject);
+            decimal? expectationsUnderlyingValue = ExtractDecimal(context.Expectation);
+
+            string subjectsName = EnumDescription(context.Subject, subjectsUnderlyingValue);
+            string expectationName = EnumDescription(context.Expectation, expectationsUnderlyingValue);
+
+            Execute.Assertion
+                .ForCondition(subject == expected)
+                .FailWith($"Expected enum to be {expectationName}{{reason}}, but found {subjectsName}.");
+        }
+
+        private static string EnumDescription(object o, decimal? v)
+        {
+            string PrintDecimal(decimal? x) => x?.ToString(CultureInfo.InvariantCulture);
```

Comment:
🔧 As `v` is null checked before calling `PrintDecimal`, it could take a regular `decimal` as parameter.